File: java/L2000/LeetCode2029.java
package L2000;

public class LeetCode2029 {
    public boolean stoneGameIX(int[] stones) {
        int[] count = new int[3];
        for (int stone : stones) {
            count[stone % 3]++;
        }
        return count[0] % 2 == 0 ? !(count[1] == 0 || count[2] == 0) : !(Math.abs(count[1] - count[2]) <= 2);
    }
}

File: cpp/L2000/LeetCode2029.cc
#include <vector>
using std::vector;

class Solution {
    bool trace(vector<int> &counts, int preSum) {

    }
public:
    bool stoneGameIX(vector<int>& stones) {
        vector<int> counts(3, 0);
        for (int s : stones) {
            counts[s % 3]++;
        }

    }
};
